### PR TITLE
Doc/dp 17060 update changelog doc

### PR DESCRIPTION
**Pull request by clairesunstudio**
This PR 
- updated changelog template and documentation
- updated changelog instructions
- fixed wrong links to changelog documentation
- fixed other small typos

### CODE REVIEW

File: changelogs/template.yml [hunk 1]
```diff
@@ -1,39 +1,59 @@
-# Write your changelog entry here.  Every PR must have a changelog.
-#___DESCRIPTION___
-#  Change_type (see below)
-#- (Project_prefix) [ComponentName] DP-1234: Adds apples to apple trees for admin apple pickers #PR
+# Every PR must have a changelog. To add a change:
```
Comment (ygannett) [inline changelogs/template.yml]: "To create a changelog" instead of "To add a change"?

File: changelogs/template.yml [hunk 2]
```diff
@@ -1,39 +1,59 @@
...
-CHANGE TYPE - [see list above]:
-  - project: [enter a valid project prefix - see list above]
-    component: [enter a valid component name - see direction above]
-    description: Describe the change. If you need multiple lines, start the first line with the following "|-" characters. (#PR)
-    issue: [add a Jira ticket or issue number, e.g. DP-12345 or 124]
-    impact: [enter a valid impact, e.g. Minor]
+Added:
+  - project: React, Patternlab
+    component: Color
+    description: Addded Color stories and color gradients. #909
```
Comment (clairesunstudio) [inline changelogs/template.yml]:
```suggestion
    description: Addded Color stories and color gradients. (#909)
```

File: changelogs/template.yml [hunk 3]
```diff
@@ -1,39 +1,59 @@
...
-#e.g. Header, Form, InputSlider, InputTextTypeAhead
+# Component name(s) in `PascalCase`
+# e.g. Header
+# e.g. Form, InputSlider, InputTextTypeAhead
+
+#___DESCRIPTION___
+# PR description and PR number (append PR number with # to create a link to the PR in Github)
+# If you need multiple lines, start the first line with the following "|-" characters.
+# For any breaking changes, add a comment in the PR describing the necessary changes on the consumer side and link that comment in the description.
+#e.g. Adds apples to apple trees for admin apple pickers #PR
```
Comment (clairesunstudio) [inline changelogs/template.yml]:
```suggestion
#e.g. Adds apples to apple trees for admin apple pickers (#PR)
```

File: changelogs/template.yml [hunk 4]
```diff
@@ -1,39 +1,59 @@
...
 # - Patch impact when you make backwards compatible bug fixes.
+# e.g. Minor
 # See https://semver.org/ for more info.
 
+
+#___TEMPLATE___
+# ChangeType [enter a valid option, see __CHANGE TYPES___ e.g. Added]:
+#  - project: Enter one or multiple (comma separated) valid project prefix(es) - see ___PROJECT___ e.g. React, Patternlab
+#    component: Enter one or multiple (comma separated) valid component name(s) - see ___COMPONENT___ e.g. Color
+#    description: Describe the change and add the PR number. see ___DESCRIPTION___ e.g. #PR
```
Comment (clairesunstudio) [inline changelogs/template.yml]:
```suggestion
#    description: Describe the change and add the PR number. see ___DESCRIPTION___ e.g. Adds apples to apple trees for admin apple pickers (#PR)
```

File: changelogs/template.yml (hunk 2 above)
Comment (clairesunstudio) [reply changelogs/template.yml]: only using () will persist in the changelog compilation

File: changelogs/template.yml [hunk 5]
```diff
@@ -1,39 +1,59 @@
-# Write your changelog entry here.  Every PR must have a changelog.
-#___DESCRIPTION___
-#  Change_type (see below)
-#- (Project_prefix) [ComponentName] DP-1234: Adds apples to apple trees for admin apple pickers #PR
+# Every PR must have a changelog. To add a change:
+# 1. Make a copy of this file and name it with the ticket number.
+# 2. Keep the example at the bottom and edit the following fields, using ___TEMPLATE___ for reference
```
Comment (ygannett) [inline changelogs/template.yml]: Suggestion: "Keep the original format of the example at the bottom, and override each element of it using ___TEMPLATE___ for reference."

File: docs/for-developers/changelog-instructions.md [hunk 6]
```diff
@@ -1,25 +1,21 @@
...
-   > This change will effect the local build. Make sure all developers know what steps to take after this gets in to dev.
+## Add a changelog for your PR (mandatory)
 
-4. For any breaking changes, add a comment in the PR describing the necessary changes on the consumer side and link that comment in the changelog.
-5. Commit the file and open your PR.
+Every PR must have a changelog file committed before getting merged into the `develop` branch. It's required for the every PR to have a changelog that comply with the format specified in [template.yml](../../changelogs/template.yml) in order to get compiled into the [CHANGELOG.md](../../CHANGELOG.md), using the [automated release script](../../scripts/compile-changelogs.js).
 
-## Release Master: When you are Releasing
+To add a changelog:
+1. make a copy of [template.yml](../../changelogs/template.yml), name it with the ticket number, e.g. `DP-1234.yml`. If there is no ticket number for the contribution, just use the branch name, e.g. `awesome-feature-branch.md`.
```
Comment (ygannett) [inline docs/for-developers/changelog-instructions.md:12]: Usually the changelog file name matches to its issue. If there is no ticket number, the issue also has the branch name?

**Review by ygannett — COMMENTED** [review]
Please see my comments below.

File: changelogs/template.yml (hunk 1 above)
Comment (clairesunstudio) [inline changelogs/template.yml]:
```suggestion
# Every PR must have a changelog. To create a changelog:
```

File: changelogs/template.yml (hunk 5 above)
Comment (clairesunstudio) [inline changelogs/template.yml]:
```suggestion
# 2. Keep the original format of the example at the bottom, and override each element of it using ___TEMPLATE___ for reference
```

File: docs/for-developers/changelog-instructions.md (hunk 6 above)
Comment (clairesunstudio) [reply docs/for-developers/changelog-instructions.md:12]: ya, I guess if there is an outside contributor or a hotfix that we don't have a ticket for, the branch name will have no ticket number

Comment (clairesunstudio) [reply docs/for-developers/changelog-instructions.md:12]: another example will be dependabot PRs

**Review by ygannett — APPROVED** [review]

**Pull request merged**